python/src/pyx/logs_db/db_mysql.py

```
@@ -1,134 +1,179 @@
     try:
-        conn = get_connection()
-        cursor = conn.cursor()
+        with get_connection() as (conn, cursor):
+            if many:
+                cursor.executemany(query, params or [])
+            else:
+                cursor.execute(query, params or ())
+            conn.commit()
+            return True
+    except MySQLError as e:
+        logger.error(f"Database query error: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)